ai2thor/controller.py

```diff
@@ -769,16 +769,20 @@ def step(self, action=None, **action_args):
 
         if not self.last_event.metadata[
             "lastActionSuccess"
-        ] and self.last_event.metadata["errorCode"] in [
+        ]:
+            if self.last_event.metadata["errorCode"] in [
             "InvalidAction",
             "MissingArguments",
             "AmbiguousAction",
             "InvalidArgument",
-        ]:
-            raise ValueError(self.last_event.metadata["errorMessage"])
+            ]:
+                raise ValueError(self.last_event.metadata["errorMessage"])
+            elif raise_for_failure:
+                raise RuntimeError(
+                    self.last_event.metadata.get("errorMessage", f"{action} failed.")
+                )
 
-        if raise_for_failure:
-            assert self.last_event.metadata["lastActionSuccess"]
+        assert (not raise_for_failure) or self.last_event.metadata["lastActionSuccess"]
```

Comment:
We should be able to remove this since the change you added will throw the exception if necessary.